### PR TITLE
[Diagnostics] Suggest platform correction based on known platforms edit distance

### LINKED ISSUE #61237: Offer fix-it when platform name in an availability attribute or query is close but not quite valid
With https://github.com/apple/swift/pull/61234 we now offer a fix-it when the case of a platform name specified in an availability attribute or query is incorrect. We could go further by diagnosing near misses using edit distance.

rdar://100241896

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
Use string similarity to suggest correct know platform in diagnostic and fix-it.

Resolves https://github.com/apple/swift/issues/61237

### CODE REVIEW

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please test

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please test

File: include/swift/AST/PlatformKind.h [hunk 1]
```diff
@@ -41,9 +41,10 @@ StringRef platformString(PlatformKind platform);
 /// or None if such a platform kind does not exist.
 Optional<PlatformKind> platformFromString(StringRef Name);
 
-/// Returns a valid platform string if the candidate string would be a valid
-/// platform string if its case were adjusted (e.g. "macos" -> "macOS").
-Optional<StringRef> caseCorrectedPlatformString(StringRef candidate);
+/// Returns a valid platform string that is closest to the candidate string
+/// based on edit distance. Returns \c None if the closest valid platform
+/// distance is not within a minimum threashold.
```
Comment (tshortli) [inline include/swift/AST/PlatformKind.h:46]:
```suggestion
/// distance is not within a minimum threshold.
```

**Review by tshortli — APPROVED** [review]
Thanks!

**Review by xymus — APPROVED** [review]
That's a nice QOL improvement, thank you!

File: include/swift/AST/PlatformKind.h (hunk 1 above)
Comment (LucianoPAlmeida) [reply include/swift/AST/PlatformKind.h]: Fixed! Thanks

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please test and merge

**Pull request merged**